packages/core/contracts/financial-templates/funding-rate-store/implementation/FundingRateStore.sol

```diff
@@ -1,8 +1,72 @@
 pragma solidity ^0.6.0;
 pragma experimental ABIEncoderV2;
 
+import "../interfaces/FundingRateStoreInterface.sol";
+import "../../../common/implementation/Testable.sol";
 
-// TODO:
-contract FundingRateStore {
 
+contract FundingRateStore is FundingRateStoreInterface, Testable {
+    struct Proposal {
+        FixedPoint.Signed rate;
+        uint256 time;
+        address proposer;
+    }
+
+    struct FundingRateRecord {
+        FixedPoint.Signed rate;
+        Proposal proposal;
+    }
+
+    enum ProposalState { None, Pending, Expired }
+
+    mapping(bytes32 => FundingRateRecord) public fundingRateRecords;
+
+    uint256 proposalLiveness;
+
+    constructor(uint256 _proposalLiveness, address _timerAddress) public Testable(_timerAddress) {
+        require(_proposalLiveness > 0, "Proposal liveness is 0");
+        proposalLiveness = _proposalLiveness;
+    }
+
+    function getFundingRateForIdentifier(bytes32 identifier) external override view returns (FixedPoint.Signed memory) {
+        FundingRateRecord storage fundingRateRecord = _getFundingRateRecord(identifier);
+
+        if (_getProposalState(fundingRateRecord.proposal) == ProposalState.Expired) {
+            return fundingRateRecord.proposal.rate;
+        } else {
+            return fundingRateRecord.rate;
+        }
+    }
+
+    function propose(bytes32 identifier, FixedPoint.Signed memory rate) external {
+        // TODO: ACLS on identifiers that can be proposed.
```

Comment:
What is ACLS?

Reply:
Access Control Lists. Just saying we need to check the whitelist to see if the identifier is approved by the DVM. Can probably make this clearer.

Reply:
Updated the comment to be clearer.